src/client/utils/searchNarratives.ts

```diff
@@ -84,107 +113,110 @@ export default async function searchNarratives({
+      fields: ['agg_fields'], // Search on all text fields
+    };
   }
+  params.filters = {
+    operator: Operator.And,
+    fields: [{ field: 'is_temporary', term: false }],
+  };
   switch (category) {
     case 'own':
-      musts.push({ term: { creator: Runtime.username() } });
-      options.auth = true;
+      params.filters.fields.push({
+        field: 'creator',
+        term: Runtime.username(),
+      });
       break;
     case 'shared':
-      musts.push({ term: { shared_users: Runtime.username() } });
-      mustNots.push({ term: { creator: Runtime.username() } });
-      options.auth = true;
+      params.filters.fields.push({
+        field: 'creator',
+        not_term: Runtime.username(),
+      });
+      params.filters.fields.push({
+        field: 'shared_users',
+        term: Runtime.username(),
+      });
       break;
     case 'public':
-      musts.push({ term: { is_public: true } });
-      options.auth = false;
+      params.access = { only_public: true };
       break;
     case 'tutorials':
-      musts.push({ term: { is_public: true } });
-      options.auth = false;
-      musts.push({
-        bool: {
-          should: [
-            { match: { narrative_title: 'tutorial' } },
-            { match: { narrative_title: 'narratorial' } },
-          ],
-        },
-      });
+      params.access = { only_public: true };
+      params.filters.fields.push({ field: 'is_narratorial', term: true });
       break;
     default:
       throw new Error('Unknown search category');
   }
 
   if (sort) {
     if (sort === 'Recently created') {
-      options.sort = [{ creation_date: { order: 'desc' } }, '_score'];
+      params.sorts = [
+        ['creation_date', SortDir.Desc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Oldest') {
-      options.sort = [{ creation_date: { order: 'asc' } }, '_score'];
+      params.sorts = [
+        ['creation_date', SortDir.Asc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Least recently updated') {
-      options.sort = [{ timestamp: { order: 'asc' } }, '_score'];
+      params.sorts = [
+        ['timestamp', SortDir.Asc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Recently updated') {
-      options.sort = [{ timestamp: { order: 'desc' } }, '_score'];
+      params.sorts = [
+        ['timestamp', SortDir.Desc],
+        ['_score', SortDir.Desc],
+      ];
     } else {
       throw new Error('Unknown sorting method');
     }
   }
-  if (skip) {
-    options.skip = skip;
-  }
-  if (musts.length) {
-    options.query.bool.must = musts;
-  }
-  if (mustNots.length) {
-    options.query.bool.must_not = mustNots;
-  }
-  return await (await makeRequest(options)).result;
+  return await (await makeRequest(params)).result;
 }
 
 /**
  *
- * @param param0 SearchOptions - this takes a query, number of documents to skip,
+ * @param param0 SearchParams - this takes a query, number of documents to skip,
```

Comment:
Should be `@param params SearchParams`

Reply:
will update :+1:

Reply:
Forgot this one, just updated

Reply:
What should it be for the `searchNarratives` function? Maybe just `@param - SearchOptions`?

Reply:
It's usually `@param parameter_name - description` So maybe `@param params - The SearchParams containing the query` or something like that?

Reply:
I fixed it to be `@param param0 - SearchOptions`

Are these comments esdoc format? Based on what I see in ESDoc, this should be something like ` @param {SearchOptions} param - search query, category, sorting, and paging for searching narratives`

There's also TSDoc which looks like it doesn't require you writing the type in the comment, which is redundant with the actual code